tests/cases/compiler/contextualPropertyOfGenericFilteringMappedType.ts

```diff
@@ -0,0 +1,33 @@
+// @strict: true
+// @noEmit: true
+
+declare function f1<T extends object>(
+  data: T,
+  handlers: { [P in keyof T as P]: (value: T[P], prop: P) => void },
+): void;
+
+f1(
+  {
+    foo: 0,
+    bar: "",
+  },
+  {
+    foo: (value, key) => {},
+    bar: (value, key) => {},
+  },
+);
+
+declare function f2<T extends object>(
+  data: T,
+  handlers: { [P in keyof T as T[P] extends string ? P : never]: (value: T[P], prop: P) => void },
```

Comment:
*Question:* Is there any reason you're using the `extends string` here instead of `PropertyKey` ?

Reply:
it's meant to actually filter something, so I intentionally filter `T[P]` that match some subset of potential types. Note that I'm checking against `T[P]` and not just `P` - but even if the check would be against `P`, I could still want to filter the keys so `PropertyKey` isn't what I'd use here